src/libs/ajax/leonardo/providers/LeoDiskProvider.ts

```diff
@@ -0,0 +1,24 @@
+import { isGcpContext } from 'src/analysis/utils/runtime-utils';
+import { Ajax } from 'src/libs/ajax';
+import { PersistentDisk } from 'src/libs/ajax/leonardo/models/disk-models';
+import { GoogleWorkspaceInfo, WorkspaceInfo } from 'src/libs/workspace-utils';
+
+export interface LeoDiskProvider {
+  list: (listArgs: Record<string, string>, signal?: AbortSignal) => Promise<PersistentDisk[]>;
+  delete: (disk: PersistentDisk, workspace: WorkspaceInfo, signal?: AbortSignal) => Promise<void>;
+}
+
+export const leoDiskProvider: LeoDiskProvider = {
+  list: (listArgs: Record<string, string>, signal?: AbortSignal): Promise<PersistentDisk[]> => {
+    return Ajax(signal).Disks.disksV1().list(listArgs);
+  },
+  delete: (disk: PersistentDisk, workspace: WorkspaceInfo, signal?: AbortSignal): Promise<void> => {
+    const { cloudContext, name, id } = disk;
+
+    if (isGcpContext(cloudContext)) {
+      const { googleProject } = workspace as GoogleWorkspaceInfo;
```

Comment:
googleProject should come from disk object not workspace!!!!

Reply:
oh no wait.... @jdcanas see my slachk msg... I remember now that this was intentional on my part because I think I found a bug in the previous code... googleProject doesn't exist on PersistentDisk type...

Reply:
ok, after digging in with @jdcanas we did indeed find a bug.  My next commit will fix this with Justin's guidance to get "goggleProject" from disk.loudContext.cloudResource, which will remove the need for workspacea arg :)

Reply:
bug fixed.  @jdcanas please include in your testing.